libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/RouteProgress.kt

```diff
@@ -184,10 +186,12 @@ data class RouteProgress(
          * Build new instance of [RouteProgress]
          *
          * @return RouteProgress
+         * @throws IllegalStateException if [DirectionsRoute] is not provided
          */
         fun build(): RouteProgress {
             return RouteProgress(
-                directionsRoute,
+                directionsRoute
+                    ?: throw IllegalArgumentException("DirectionsRoute is required to build a RouteProgress."),
```

Comment:
nit: 

check(directionsRoute != null) { "DirectionsRoute is required to build a RouteProgress." }

Reply:
```kotlin
requireNotNull(directionsRoute) { "DirectionsRoute is required to build a RouteProgress." }
```